paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CloneAndMigrateAction.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.action;
+
+import org.apache.paimon.catalog.Identifier;
+import org.apache.paimon.flink.clone.CloneAndMigrateUtils;
+import org.apache.paimon.flink.sink.FlinkStreamPartitioner;
+import org.apache.paimon.options.CatalogOptions;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.functions.sink.v2.DiscardingSink;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Clone External source files and commit metas to construct Paimon table. Currently, supports:
+ *
+ * <p>Hive table to paimon table.
+ */
+public class CloneAndMigrateAction extends ActionBase {
+
+    private final Map<String, String> sourceCatalogConfig;
+    private final String sourceDatabase;
+    private final String sourceTableName;
+
+    private final Map<String, String> targetCatalogConfig;
+    private final String targetDatabase;
+    private final String targetTableName;
+
+    private final int parallelism;
+
+    public CloneAndMigrateAction(
+            String sourceDatabase,
+            String sourceTableName,
+            Map<String, String> sourceCatalogConfig,
+            String targetDatabase,
+            String targetTableName,
+            Map<String, String> targetCatalogConfig,
+            @Nullable Integer parallelism) {
```

Comment:
We should have the ability to specify partitions, and even the ability to specify formats such as `dt>20250303`.